File: src/main/java/com/aif/associations/controller/AssociationsExtractor.java
package com.aif.associations.controller;

import com.aif.associations.model.IAssociationsExtractor;
import com.aif.associations.model.IConnectionExtractor;
import com.aif.associations.model.IItem;

import javax.validation.constraints.Max;
import javax.validation.constraints.Min;

public class AssociationsExtractor implements IAssociationsExtractor {

    @Min(value = 0)
    @Max(value = 1)
    @Override
    public double getAssociationsLevel(IConnectionExtractor connectionExtractor, IItem item1, IItem item2) {
        return 0;
    }

}

File: src/main/scala/com/aif/associations/model/AssociationsExtractor.scala
package com.aif.associations.model

import javax.validation.constraints.{Max, Min}

trait AssociationsExtractor[T <: Item, T2 <: Item]{

  @Min(value = 0)
  @Max(value = 1)
  def getAssociationsLevel(connectionExtractor: ConnectionExtractor[T, T2], item1: T, item2: T2): Double
}
